### PR TITLE
Kobo Elipsa support

### LINKED ISSUE #7890: [NEW DEVICE] Does not launch on Kobo elipsa
* KOReader version: 2021.06
* Device: Kobo Elipsa

#### Issue

The application does not launch on the new Kobo elipsa. The device's stock interface freezes after I try to launch KOReader from nickel menu. The device UI eventually restarts.

#### Steps to reproduce

Install KOReader and nickel menu on the new Kobo elipsa. Click on Nickel Menu > KOReader. The device will freeze and then eventually restart.

##### `crash.log`
[crash.log](https://github.com/koreader/koreader/files/6719564/crash.log)

### PR DESCRIPTION
**Pull request by NiLuJe**
Requires https://github.com/koreader/koreader-base/pull/1386

Fix #7890 

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7986)

### CODE REVIEW

**Comment by NiLuJe** [comment]
TODO:

* Better pen handling (right now, the sheer amount of events and lack of distinction between hovering and pen down tend to make pen input trip hold timers).
* Testing multi-touch

**Review by Frenzie — APPROVED** [review]
Looks fine (btw, wrt shellcheck don't forget to see what new versions do :-) )

File: frontend/device/kobo/device.lua [hunk 1]
```diff
@@ -803,6 +853,14 @@ function Kobo:toggleChargingLED(toggle)
     if toggle == true then
         -- NOTE: Technically, Nickel forces a toggle off before that, too.
         --       But since we do that on startup, it shouldn't be necessary here...
+        if self:isSunxi() then
+            f:write("ch 3")
+            f:flush()
+            f:write("cur 1")
+            f:flush()
+            f:write("dc 63")
```
Comment (poire-z) [inline frontend/device/kobo/device.lua:861]: Some comments about what each of these commands do (if you could guess) ?

File: platform/kobo/koreader.sh [hunk 2]
```diff
@@ -241,6 +241,36 @@ if [ -z "${INTERFACE}" ]; then
...
+    KOBO_TS_INPUT="/dev/input/event1"
+fi
+
+# Make sure we only have only two cores online on the Elipsa.
+# NOTE: That's a bit optimistic, we might actually need to tone that down to one,
+#       and just online the second one on demand (e.g., PDF).
+if [ "${PRODUCT}" = "europa" ]; then
+    echo "1" >"/sys/devices/system/cpu/cpu1/online"
+    echo "0" >"/sys/devices/system/cpu/cpu2/online"
+    echo "0" >"/sys/devices/system/cpu/cpu3/online"
```
Comment (poire-z) [inline platform/kobo/koreader.sh:271]: >  `# Make sure we only have only two cores online `

But you _did_ tore that down to one ? Or echo "1" means disable ?
(Will cpu1 core suffers if it's the only one getting used ? :)

**Review by poire-z — APPROVED** [review]

File: frontend/device/kobo/device.lua (hunk 1 above)
Comment (NiLuJe) [reply frontend/device/kobo/device.lua:861]: Done ;).

File: platform/kobo/koreader.sh (hunk 2 above)
Comment (NiLuJe) [reply platform/kobo/koreader.sh:271]: CPUs start counting at 0, and you can't offline CPU0 (for obvious reasons ;p).

Comment (poire-z) [reply platform/kobo/koreader.sh:271]: Ok :) (Too much Lua and counting from 1!)

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 2 of 8 files at r1, 4 of 5 files at r2, 3 of 3 files at r3.
> Dismissed @poire-z from a discussion.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/7986)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @pazos)

___
*[frontend/device/kobo/device.lua, line 861 at r2](https://reviewable.io/reviews/koreader/koreader/7986#-Mer-Z1CGVNA2MHvNrIP-r2-861:-Mf8ZU2S6m65Zxl0PhIb:b-896fix) ([raw file](https://github.com/koreader/koreader/blob/01afefd0604eb7c12b70c4f5ef8a3437a3057627/frontend/device/kobo/device.lua#L861)):*
<details><summary><i>Previously, NiLuJe (NiLuJe) wrote…</i></summary><blockquote>

Done ;).
</blockquote></details>

Done.

**Review by NiLuJe — COMMENTED** [review]
> Reviewed 1 of 1 files at r4.
> *[Reviewable](https://reviewable.io/reviews/koreader/koreader/7986)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @pazos)

**Pull request merged**